src/controllers/todoController.ts

```diff
@@ -0,0 +1,62 @@
+import { TodoService } from "../services/todoService";
+import { Request, Response } from "express";
+
+const NO_ID_MESSAGE = 'Mission Id required!!';
+const LIST_IS_EMPTY = "todo's list is empty!";
+export class TodoController {
+    constructor(private todoService: TodoService) {}
+
+    getTodos = (req: Request, res: Response) =>{
+        const todos = this.todoService.read();
+        if (!todos){
+            res.status(400).json('failed to Get missions')
+            return;
+        }
+        res.status(200).json(todos);   
+    }
+
+    createNewTodo = (req: Request, res: Response) =>{
+        const {name, date} = req.body;
+        if(!name || !date){
+            res.status(400).json({error:'Mission and Date are required!!'});
+            return;
+        }
+        if(isNaN(Date.parse(date))){
+            res.status(400).json({error:"Date is invalid! try again"});
+            return;
+        }
+        const dateInDateFormat = new Date(date);
+        dateInDateFormat.setHours(12, 0, 0, 0)
+        const itemToReturn = this.todoService.add(name, dateInDateFormat);
+        res.status(200).json(itemToReturn);
+    }
+
+    deleteTodo = (req: Request, res: Response) =>{
+        const missionIdToDelete = req.params.id;
+        if(!missionIdToDelete){
+            res.status(400).json({error: NO_ID_MESSAGE})
+            return;
+        }
+        const itemToReturn = this.todoService.delete(missionIdToDelete);
+        res.status(200).json(itemToReturn);
+    }
+
+    toggleTodo = (req: Request, res: Response) =>{
+        const id = req.params.id;
+        if(!id){
+            res.status(400).json({error:NO_ID_MESSAGE})
+            return;
+        }
+        const itemToReturn = this.todoService.toggle(id);
+        res.status(200).json(itemToReturn);
```

Comment:
I know we didn't cover it up. but that's a good opportunity
The response number  typically for successful delete will be 204, successful create will be 201